packages/react-native-views/src/components/CameraView/hooks/usePictures.js

```diff
@@ -0,0 +1,35 @@
+import { useCallback, useState } from 'react';
+import noop from 'lodash.noop';
+
+import useSights from './useSights';
+
+function usePictures(camera, sights, onTakePicture, handleFakeActivity = noop) {
+  const { activeSight, count: nbOfSights, nextSightProps } = useSights(sights);
+  const [pictures, setPictures] = useState({});
+
+  const handleTakePicture = useCallback(async () => {
+    if (camera) {
+      handleFakeActivity();
+
+      const options = { quality: 1 };
+      const picture = await camera.takePictureAsync(options);
+      const payload = { sight: activeSight, source: picture };
+
+      setPictures((prevState) => ({ ...prevState, [activeSight.id]: payload }));
+      onTakePicture(payload);
+
+      if (!nextSightProps.disabled) {
+        nextSightProps.onPress();
+      }
+    }
```

Comment:
I usually like to change this kind of patterns into 
```js
if (!camera) return;

...
```
It reduces indentation and clearly shows that not having a camera instantly exits.